### PR TITLE
update golang and ocs-operator dependency version

### PR DESCRIPTION
**Pull request by leelavg**
- Update golang and ocs-operator dependency version
- Fix function signatures/usage for updated kube packages

### CODE REVIEW

**Comment by leelavg** [comment]
Reason for the PR to be in Draft state:
- Replace directive with local path (depends on https://github.com/red-hat-storage/ocs-operator/pull/2195)

CI Failures:
- I'll relook at a later point of time as I see lint & build is passing for now locally
```
# date && go version && make lint build && date
Wednesday 27 September 2023 09:23:22 AM IST
go version go1.20.8 linux/amd64
docker run --rm -v /root/toolbox/orc/ocs-client-operator:/app:Z -w /app golangci/golangci-lint:v1.54.2 golangci-lint run ./...
level=warning msg="[runner] The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused."
level=warning msg="[runner] The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused."
level=warning msg="[runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused."
go mod tidy && go mod vendor
/root/toolbox/orc/ocs-client-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
docker build -t quay.io/ocs-dev/ocs-client-operator:latest .
[+] Building 230.8s (23/23) FINISHED                                                                                                             docker:default
 => [internal] load .dockerignore                                                                                                                          0.1s
 => => transferring context: 171B                                                                                                                          0.0s
 => [internal] load build definition from Dockerfile                                                                                                       0.1s
 => => transferring dockerfile: 921B                                                                                                                       0.0s
 => [internal] load metadata for gcr.io/distroless/static:nonroot                                                                                          0.0s
 => [internal] load metadata for docker.io/library/golang:1.20                                                                                             2.7s
 => [auth] library/golang:pull token for registry-1.docker.io                                                                                              0.0s
 => [builder  1/13] FROM docker.io/library/golang:1.20@sha256:6b29720c5598cb40b23dd38f522050a698d427c3a5696c5efe1ed22bd45b1396                             0.0s
 => CACHED [stage-1 1/4] FROM gcr.io/distroless/static:nonroot                                                                                             0.0s
 => [internal] load build context                                                                                                                          1.9s
 => => transferring context: 51.06MB                                                                                                                       1.9s
 => CACHED [builder  2/13] WORKDIR /workspace                                                                                                              0.0s
 => CACHED [builder  3/13] COPY go.mod go.sum ./                                                                                                           0.0s
 => CACHED [builder  4/13] COPY vendor/ vendor/                                                                                                            0.0s
 => CACHED [builder  5/13] COPY main.go Makefile ./                                                                                                        0.0s
 => [builder  6/13] COPY hack/ hack/                                                                                                                       1.1s
 => [builder  7/13] COPY api/ api/                                                                                                                         0.2s
 => [builder  8/13] COPY controllers/ controllers/                                                                                                         0.1s
 => [builder  9/13] COPY config/ config/                                                                                                                   0.2s
 => [builder 10/13] COPY pkg/ pkg/                                                                                                                         0.2s
 => [builder 11/13] COPY service/ service/                                                                                                                 0.1s
 => [builder 12/13] RUN make go-test                                                                                                                     101.0s
 => [builder 13/13] RUN make go-build                                                                                                                    120.4s
 => [stage-1 2/4] COPY --from=builder /workspace/bin/manager .                                                                                             0.1s
 => [stage-1 3/4] COPY --from=builder /workspace/bin/status-reporter .                                                                                     0.1s
 => exporting to image                                                                                                                                     1.1s
 => => exporting layers                                                                                                                                    1.0s
 => => writing image sha256:9fc90904914467c5f07c4b55ef41bb604c3ba8fec95c664a7900337429043c76                                                               0.0s
 => => naming to quay.io/ocs-dev/ocs-client-operator:latest                                                                                                0.0s
Wednesday 27 September 2023 09:29:23 AM IS

```

File: go.mod [hunk 1]
```diff
@@ -1,91 +1,90 @@
 module github.com/red-hat-storage/ocs-client-operator
 
-go 1.18
+go 1.20
 
 replace (
 	github.com/portworx/sched-ops => github.com/portworx/sched-ops v0.20.4-openstorage-rc3 // required by Rook v1.12
+	k8s.io/api => k8s.io/api v0.26.4
```
[the remark this answers is not in the document]
Comment (leelavg) [reply go.mod:7]: updated, the reason was:
- there were inconsistencies with modules like gnostic, kube-openapi, k8s.io/api and this was the version without any updates to existing `require` directives that was working with replace of ocs-operator to local path earlier
- after removing replace directive and update of deps is working, I believe `client-go incompatible` in `require` might have caused the issue in the first place.

File: go.mod [hunk 2]
```diff
@@ -1,91 +1,90 @@
...
-	google.golang.org/grpc v1.53.0
+	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.67.1
+	github.com/red-hat-storage/ocs-operator/v4 v4.0.0-20230915072501-d290a4270291
+	github.com/stretchr/testify v1.8.4
+	google.golang.org/grpc v1.56.2
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.26.4
-	k8s.io/apimachinery v0.26.4
+	k8s.io/api v0.28.0
+	k8s.io/apimachinery v0.28.0
```
[the remark this answers is not in the document]
Comment (leelavg) [reply go.mod]: same as above, updated now.

**Review by Madhu-1 — COMMENTED** [review]
changes LGTM, @leelavg any specific reason to get this merged to the custom 4.14 branch, I would suggest getting this merged to main branch and backport it to the custom branch.

**Comment by leelavg** [comment]
I was asked (by Ohad) to have every commit for hci effort across all the repos that we test to this custom branch and build team would use that branch.

**Pull request merged**